### ISSUE TITLE
Set `Type` in the `ErrorModel`

### ISSUE THREAD

**Issue by victoraugustolls** [danielgtaylor/huma#307]
Right now, I see no way to directly set the `Type` of the error as the interface `StatusError` doesn't include a method, and `huma.NewError` also doesn't accept a `type` parameter! Maybe there is something obvious that I'm missing, but right now, I'm passing a custom error type to `NewError` just to set this field!

@danielgtaylor WDYT of maybe creating a `NewErrorWithType` or adding a method for `StatusError` to set the type?

**Comment by danielgtaylor** [comment]
@victoraugustolls I don't understand, maybe a code example of what you are trying to do would help? You should be able to just create a `huma.ErrorModel` instance with whatever `Type` field value you want and return it as the error. You can create your own `NewErrorWithType` to wrap it all up in your own codebase as needed.

`huma.NewError` is just a convenience function and it deals with a generic "error with HTTP status" concept. Those errors don't have to be RFC9457 and won't necessarily have a `Type` field.

**Comment by victoraugustolls** [comment]
I understand that the errors don't need to be RFC9457! But if it is, the `NewError` does not provide a parameter to inform the type. What I'm doing right now is:

```go
type ErrorModel struct {
	huma.ErrorModel
}

func (e *ErrorModel) ContentType(ct string) string {
	if ct == "application/json" {
		return "application/json"
	}
	return ct
}

type TypeErr string

func (e TypeErr) Error() string {
	return string(e)
}

var NewErrorWithType = func(status int, msg string, errs ...error) huma.StatusError {
	var details []*huma.ErrorDetail

	var (
		tp      string
		typeErr TypeErr
	)
	for i := 0; i < len(errs); i++ {
		if errs[i] == nil {
			continue
		}

		if converted, ok := errs[i].(huma.ErrorDetailer); ok {
			details = append(details, converted.ErrorDetail())
		} else if errors.As(errs[i], &typeErr) {
			tp = typeErr.Error()
		} else {
			details = append(details, &huma.ErrorDetail{Message: errs[i].Error()})
		}
	}
	return &ErrorModel{
		huma.ErrorModel{
			Type:   tp,
			Status: status,
			Title:  http.StatusText(status),
			Detail: msg,
			Errors: details,
		},
	}
}
```

**Comment by victoraugustolls** [comment]
If that is the expected behavior, that's ok! But I feel `huma` should provide a way to set the type because it exists by default.

**Comment by danielgtaylor** [comment]
@victoraugustolls I do wonder if we are overcomplicating this. Compare it to:

```go
err := huma.Error403Forbidden("invalid action for token", err)
err.Type = "srn:error:unauthorized"
```

This way you can also set the `Instance` field and others as needed without any "magic" behind the scenes. If it becomes repetitive you wrap it in a simple utility function. What do you think?

**Comment by victoraugustolls** [comment]
Also, the only reason that I am wrapping the `huma.ErrorModel` is to change the default content type. Is there an easier way to do this?

**Comment by danielgtaylor** [comment]
@victoraugustolls yeah you would have to cast, which is a bit annoying to be honest, I guess like:

```go
err := huma.Error403Forbidden("invalid action for token", err).(*huma.ErrorModel)
err.Type = "srn:error:unauthorized"
```

Alternatively you can write your own utility functions for the different error responses.

> Also, the only reason that I am wrapping the `huma.ErrorModel` is to change the default content type. Is there an easier way to do this?

No, this looks like a reasonable way to do it. :+1:

**Comment by victoraugustolls** [comment]
@danielgtaylor  do you feel like there should be an easier way to do both of this operations? If so, I can try of think of something non breaking and suggest it here.

**Comment by danielgtaylor** [comment]
Moving this to a discussion until we can think up some ways to make it work. In the meantime the easiest solution is probably just to create your own utility methods that behave like `huma.Error*` but set all the fields you want.
